tests/integration/test_manifest_validation.py

```diff
@@ -0,0 +1,487 @@
+    raise EnvironmentError("SYNAPSE_PAT environment variable is not set.")
+
+HEADERS = {"Authorization": f"Bearer {SYNAPSE_PAT}"}
+EXAMPLE_SCHEMA_URL = "https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.model.jsonld"
+
+@pytest.fixture(scope="module")
+def setup_api(testing_config: ConfigurationForTesting) -> str:
+    url = testing_config.schematic_api_server_url
+    ui_endpoint = os.path.join(url, "v1/ui")
+
+    # Make a request to the API and make sure it is responsive
+    response = requests.get(ui_endpoint)
+
+    assert response.status_code == 200, f"Failed to connect to API: {response.text}"
+    return url
+
+class TestManifestValidation:
+
+    @pytest.mark.parametrize(
+        ("input_data_type", "input_file_name"),
+        [
+            ("Biospecimen", "mock_manifests/example_biospecimen_test.csv"),
+            ("Patient", "mock_manifests/example.patient_component_rule.manifest.csv"),
+        ],
+    )
+    def test_manifest_validation_basic_valid(
+        self, input_data_type: str, input_file_name: str, setup_api, helpers: Helpers
+    ) -> None:
+        """
+        Test that the manifest validation API returns no errors when a valid manifest is provided.
+
+        Arguments:
+            input_data_type: The data type of the manifest
+            input_file_name: The name of the manifest file
+            setup_api: A pytest fixture returning the API endpoint to use
+            helpers: The Helpers object
+
+        Returns:
+            None
+        """
+        # GIVEN the manifest validation endpoint and parameters
+        url = os.path.join(setup_api, "v1/model/validate")
+        print(url)
+        params = {
+            "schema_url": EXAMPLE_SCHEMA_URL,
+            "data_type": input_data_type,
+            "data_model_labels": "class_label",
+            "restrict_rules": False,
+        }
+
+        # AND a valid file
+        file_path = helpers.get_data_path(input_file_name)
+        with open(file_path, "rb") as file_obj:
+            files = {"file_name": (os.path.basename(file_path), file_obj, "text/csv")}
+
+            # WHEN we make a POST request to validate the file
+            response = requests.post(url, params=params, files=files)
+
+            # THEN we expect a successful response
+            assert (
+                response.status_code == 200
+            ), f"Got status code: {response.status_code}. Expected '200'."
+
+            # AND with expected keys in the json
+            response_json = response.json()
+            assert "warnings" in response_json.keys()
+            assert "errors" in response_json.keys()
+
+            # AND with no expected errors
+            assert len(response_json.get("errors")) == 0
+
+    @pytest.mark.parametrize(
+        ("input_data_type", "input_file_name"),
+        [
+            (
+                "Biospecimen",
+                "mock_manifests/Invalid_Biospecimen_Missing_Column_Manifest.csv",
```

Comment:
@jaymedina if this is not too much work, could you switch to use the invalid manifest that I provided in the test plan. I saw that this manifest: Invalid_Biospecimen_Missing_Column_Manifest.csv is already being used by test: `test_validation.py::TestManifestValidation::test_missing_column`.. and this error message was already tested there. I think if you use the invalid manifest that you provided in the test plan, it could also help us better "map" the manual tests with integration tests. 

Thank you!

Reply:
Updated!